tools/mpremote/mpremote/commands.py

```diff
@@ -352,7 +365,13 @@ def do_filesystem(state, args):
             elif command == "mkdir":
                 state.transport.fs_mkdir(path)
             elif command == "rm":
-                state.transport.fs_rmfile(path)
+                if args.recursive:
+                    if path:
+                        do_filesystem_recursive_rm(state, path, args)
+                    else:
+                        raise CommandError("rm: missing path")
```

Comment:
I don't think it's possible to get to this line of code, the argument parser should guarantee that path is specified, right?

Reply:
If we want to support `mpremote rm -r :` then this check for path being non-zero is not needed.

Reply:
You are correct, overly cautious 

partly due the guard above: 
```
    # ls implicitly lists the cwd.
    if command == "ls" and not paths:
        paths = [""]
```